redwoods-insurance-salesforce/force-app/main/default/lwc/incidentAudioPlayer/__tests__/incidentAudioPlayer.test.js

```diff
@@ -0,0 +1,65 @@
+import { createElement } from 'lwc';
+import IncidentAudioPlayer from 'c/incidentAudioPlayer';
+import { registerApexTestWireAdapter } from '@salesforce/sfdx-lwc-jest';
+import { getRecord } from 'lightning/uiRecordApi';
+import { getAudio } from '@salesforce/apex/IncidentController.findRelatedFiles';
+import { registerLdsTestWireAdapter } from '@salesforce/sfdx-lwc-jest';
+
+const mockAudioFile = require('./data/mockAudioFile.json');
+const getRelatedAudioAdapter = registerApexTestWireAdapter(getAudio);
+const getRecordAdapter = registerLdsTestWireAdapter(getRecord);
+
+describe('c-incident-audio-player', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    it('does not show the player when no audio file is provided.', () => {
+        // Create initial element
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        document.body.appendChild(element);
+
+        const labelElement = element.shadowRoot.querySelector('p');
+        expect(labelElement.textContent).toContain(
+            'There are currently no audio files for this case.'
+        );
+    });
+
+    it('displays the audio player when the wire adapter returns an audio file', () => {
+        const element = createElement('c-incident-audio-player', {
+            is: IncidentAudioPlayer
+        });
+        document.body.appendChild(element);
+
+        getRecordAdapter.emit('mockRecordId');
+        getRelatedAudioAdapter.emit(mockAudioFile);
+
+        return Promise.resolve().then(() => {
+            const audioPlayerEl = element.shadowRoot.querySelector('audio');
+            expect(audioPlayerEl).not.toBe(null);
```

Comment:
We should also test that the `key` and `src` attributes get the value of `url` passed.

Reply:
It should be tested that `AUDIO` is set as `fileType` on the wire call (using `getLastConfig`).

Reply:
@muenzpraeger - KEY is not an available property to inspect. Here's a JSON representation of the audio tag:

```json
{"nodeValue":null,"textContent":"Your browser doesn't support audio tags.","innerHTML":"<code>Your browser doesn't support audio tags.</code>","outerHTML":"<audio src=\"/sfc/servlet.shepherd/version/download/file1\" type=\"audio/x-m4a\" controls=\"true\" class=\"slds-large-size_1-of-1\"><code>Your browser doesn't support audio tags.</code></audio>"}
```

URL on the other hand, is.

Reply:
Wooops. The `iterator` key... 🤦  has been a long day.